src/audio/module_adapter/module/volume/volume.c

```diff
@@ -678,18 +667,18 @@ static inline int volume_set_chan(struct processing_module *mod, int chan,
 	 * multiplication overflow with the 32 bit value. Non-zero MIN option
 	 * can be useful to prevent totally muted small volume gain.
 	 */
-	if (v < VOL_MIN) {
+	if (v < cd->vol_min) {
 		/* No need to fail, just trace the event. */
 		comp_warn(mod->dev, "volume_set_chan: Limited request %d to min. %d",
-			  v, VOL_MIN);
-		v = VOL_MIN;
+			  v, cd->vol_min);
+		v = cd->vol_min;
```

Comment:
there are other place also set vol_min and max, we are better to check condition is possible, if not possible, I would suggest remove the check.

Reply:
Yes possibly, this is not speed critical code. I'll see if this can be improved.

Reply:
> 

0.5 MCPS, how this calculated out? this patch more focused on code refine, I don't think there is 0.5 mcps improvement. window ramp itself cost about 0.9 as I measured without this PR.

Reply:
The 0.5 MCPS was from that time 1 ms copy() length traced timestamps. Now with more code in-lined the improvement seems to be 0.9 MCPS same way.  I tried also perf counters but it's not suitable  to check a 20 ms long operation mode. If I would change in topology ramp to longer the load would be lower because the ramp update is less frequent (1/2 or 1/4) with the less steep ramps. It seems that the overhead of 1 ms traces is quite large. I get base 4.3 - 4.5 MCPS that way for volume while perf counters give 2.3 - 2.6 MCPS. I assumed that perf counters measure 38.4 MHz timer ticks and that DSP clock in TGL-Zephyr-IPC3 is 400 MHz. Is that correct?

Reply:
Also please check in the bottom of this discussion the plots, there's the evidence of MCPS saving.

Reply:
yes, this is due to log print takes a lot of time, if you use 1ms to measure, you will get overestimated data, 1024ms is better, maybe in future, we can continue enlarge it to 10240ms for more accurate data, log print is really a big overhead to sof.